### DISCUSSION TITLE
What is the best practice for useId and getting rid of these hydration errors.

### DISCUSSION THREAD

**Discussion by pixelpaulaus** [nuxt/nuxt#27049] [Questions]
So i have been getting the missmatch hydration errors since the beginning of the year, and there is a lot of talk about using certain vue versions, and waiting for vue 3.5 and so on. I am using headless UI and NuxtUI btw.
And then headless UI release a fix which is....

```
import { provideUseId } from '@headlessui/vue'
provideUseId(() => useId())
```

But now I am getting the errors....

`[nuxt] `useId` might not work correctly with components that have `inheritAttrs: false`.`

Which there is tasks and bug reports about. But I can find a clear answer for a fix.
Does anyone know or have any suggestions? or is it safe to ignore?

**Comment by marocama** [comment]
Any solution for this problem?

I'm defining in my `app.vue`:
`provideHeadlessUseId(() => useId())`

But I have the same problem, using Headless UI

**Comment by davidabou** [comment]
Same issue

**Comment by simonmaass** [comment]
could this solve this?: https://blog.vuejs.org/posts/vue-3-5#useid

**Comment by r-ijal** [comment]
Anyone solved this?
